golang/cosmos/x/swingset/genesis.go

```diff
@@ -34,13 +32,37 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data *types.GenesisState) bool
 	keeper.SetParams(ctx, data.GetParams())
 	keeper.SetState(ctx, data.GetState())
 
+	swingStoreExportData := data.GetSwingStoreExportData()
+	if len(swingStoreExportData) > 0 {
+		// See https://github.com/Agoric/agoric-sdk/issues/6527
+		panic("genesis with swing-store state not implemented")
+	}
+
 	// TODO: bootstrap only if not restoring swing-store from genesis state
 	return true
 }
 
 func ExportGenesis(ctx sdk.Context, k Keeper) *types.GenesisState {
-	gs := NewGenesisState()
-	gs.Params = k.GetParams(ctx)
-	gs.State = k.GetState(ctx)
+	gs := &types.GenesisState{
+		Params:               k.GetParams(ctx),
+		State:                k.GetState(ctx),
+		SwingStoreExportData: []*types.SwingStoreExportDataEntry{},
+	}
+
+	// Only export the swing-store shadow copy for now
+	// TODO:
+	// - perform state-sync export with check blockHeight (figure out how to
+	//   handle export of historical height),
+	// - include swing-store artifacts in genesis state
+	// See https://github.com/Agoric/agoric-sdk/issues/6527
```

Comment:
What is our current stance with respect to genesis export/import? This PR modifies `InitGenesis` to reject input that includes swingStore export data while also modifying `ExportGenesis` to include such data, which has the benefit of not losing it but the (new?) drawback of guaranteeing that we cannot consume our own exports without modifying them. I guess that's a reasonable position to be in for now, but the comments should probably be more explicit about it.

Reply:
Genesis export will be addressed in an upcoming PR building on the foundation laid here. Right now I'm just replicating the existing status where only the export data is in genesis, but failing at import if any swing store export data is present to avoid the confusion that would happen today.